mne/io/snirf/_snirf.py

```diff
@@ -531,49 +547,48 @@ def _get_lengthunit_scaling(length_unit):
         )
 
 
-def _extract_sampling_rate(dat):
+def _extract_sampling_rate(dat, user_sfreq):
     """Extract the sample rate from the time field."""
     # This is a workaround to provide support for Artinis data.
     # It allows for a 1% variation in the sampling times relative
     # to the average sampling rate of the file.
     MAXIMUM_ALLOWED_SAMPLING_JITTER_PERCENTAGE = 1.0
 
+    _validate_type(user_sfreq, ("numeric", None), "sfreq")
     time_data = np.array(dat.get("nirs/data1/time"))
-    sampling_rate = 0
-    if len(time_data) == 2:
-        # specified as onset, samplerate
-        sampling_rate = 1.0 / (time_data[1] - time_data[0])
```

Comment:
this was a bug. length-2 `time_data` is `(onset, period)` so the correct sampling rate would have been `1.0 / time_data[1]`